instrumentation/spring/spring-batch-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/batch/chunk/ChunkExecutionInstrumenter.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.chunk;
+
+import static io.opentelemetry.javaagent.instrumentation.api.Java8BytecodeBridge.currentContext;
+import static io.opentelemetry.javaagent.instrumentation.spring.batch.SpringBatchInstrumentationConfig.instrumentationName;
+import static io.opentelemetry.javaagent.instrumentation.spring.batch.SpringBatchInstrumentationConfig.shouldCreateRootSpanForChunk;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanLinksBuilder;
+import org.springframework.batch.core.scope.context.ChunkContext;
+import org.springframework.batch.core.step.builder.SimpleStepBuilder;
+
+public class ChunkExecutionInstrumenter {
+
+  private static final Instrumenter<ChunkContextAndBuilder, Void> INSTRUMENTER =
+      Instrumenter.<ChunkContextAndBuilder, Void>newBuilder(
+              GlobalOpenTelemetry.get(),
+              instrumentationName(),
+              ChunkExecutionInstrumenter::spanName)
+          .addSpanLinksExtractor(ChunkExecutionInstrumenter::extractSpanLinks)
+          .newInstrumenter();
+
+  public static Instrumenter<ChunkContextAndBuilder, Void> chunkExecutionInstrumenter() {
+    return INSTRUMENTER;
+  }
+
+  private static void extractSpanLinks(
+      SpanLinksBuilder spanLinks, Context unused, ChunkContextAndBuilder request) {
+    // The context passed will be Context.root() if shouldCreateRootSpanForChunk()
+    Context parentContext = currentContext();
```

Comment:
This is a bit strange. Rather than using the context passed as an argument, I need to get the current context. This is necessary because `TracingChunkExecutionListener#beforeChunk` needs to conditionally set the span as a root span based off `shouldCreateRootSpanForChunk`. The only way (as far as I know) for `TracingChunkExecutionListener#beforeChunk` to start a span as a root span is by passing `Context.root()` as the parent context. 

But if `Context.root()` is passed as the parent context, then this span link extractor code will not be able to access the original parent context necessary to add a span link.

Reply:
I don't think there's a better way to solve that - your version is perfectly fine.